src/hooks/useWeb3ReactCore.ts

```diff
@@ -0,0 +1,37 @@
+import { ChainId } from '@swapr/sdk'
+
+import { useWeb3React, Web3ContextType } from '@web3-react/core'
+import { providers } from 'ethers'
+import { useEffect, useState } from 'react'
+
+import { web3Network } from '../connectors'
+import { NETWORK_DETAIL } from '../constants'
+import getLibrary from '../utils/getLibrary'
+
+type Web3ReactProps = Omit<Web3ContextType, 'chainId'> & {
+  chainId?: ChainId
+  isCurrentChainUnsupported: boolean
+}
+
+let networkLibrary: providers.Web3Provider | undefined
+export function getNetworkLibrary(): providers.Web3Provider {
+  return (networkLibrary = networkLibrary ?? getLibrary(web3Network.provider))
+}
```

Comment:
check if you need it. `library` was used in previous version

Reply:
fixed [706972b](https://github.com/levelkdev/swapr-dapp/pull/1330/commits/706972b3541298f5732288bf2287389aec894d61)